src/Core/Framework/Bundle.php

```diff
@@ -115,17 +115,18 @@ protected function getServicesFilePath(): string
         return 'Resources/config/services.xml';
     }
 
-    protected function registerFilesystem(ContainerBuilder $container, string $key): void
+    protected function registerFilesystem(ContainerBuilder $container, string $key, ?string $baseKey = null): void
     {
         $containerPrefix = $this->getContainerPrefix();
-        $parameterKey = sprintf('shopware.filesystem.%s', $key);
+        $parameterKey = sprintf('shopware.filesystem.%s', $baseKey ?? $key);
```

Comment:
with this you will set the reference to a non existing file system. because the parameterKey will be something like `shopware.filesystem.my_custom_filesystem`

Reply:
No. That is the point of my change. When I choose something like `registerFilesystem($c, 'party')` I try to get a non existing file system. As of my change I can write `registerFilesystem($c, 'party', 'private')` and get an existing file system (`shopware.filesystem.private` in this case).

Reply:
this parameter order is very confusing then. normally the second parameter is always `public` or `private` but not, if I want to provide a custom name, than it is the third parameter?

Reply:
I understand why it can be confusing. My intentional thought about it was like: the second parameter is the name my file system gets and the third parameter what it is based on.

Any suggestion how to remove the confusion?